src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
```

Comment:
The storage map stores a MapPair per Entry. That consumes memory in almost as significant amounts as the Entry does due to the way key strings may be longer than the Entry data string coming out of helpers. Hopefully the recent alterations make that more clear.